typescript/free-first-mint/pages/index.tsx

```diff
@@ -1,21 +1,32 @@
 import { useAddress, useMetamask, useSignatureDrop } from "@thirdweb-dev/react";
 import type { NextPage } from "next";
+import { useState } from "react";
 
 const Home: NextPage = () => {
   const address = useAddress();
   const connectWithMetamask = useMetamask();
-  const drop = useSignatureDrop("0xcB31341eE7FaC6917e8e9D71441747e5FAdA466F");
+  const signatureDrop = useSignatureDrop(
+    "0x6d148a12f7c0ae693609F5a26E085646f8F73A53"
+  );
+  const [quantity, setQuantity] = useState<number>(1);
 
   const mint = async () => {
     const signedPayloadReq = await fetch("/api/generate-mint-sig", {
       method: "POST",
-      body: JSON.stringify({ address }),
+      body: JSON.stringify({ address, quantity }),
     });
-
     const signedPayload = await signedPayloadReq.json();
 
     try {
-      const nft = await drop?.signature.mint(signedPayload.signedPayload);
+      const nft = await signatureDrop?.signature.mint(
+        signedPayload.signedPayload
+      );
+      if (nft) {
```

Comment:
this being in a different request makes no sense, since I can just not send this request and then claim for free forever, this should be in the generate signature minting request, you generate one signature to claim for free, if the user don't use it, they lose it